controllers/clustergroupupgrade_controller.go

```diff
@@ -2112,6 +2115,19 @@ func (r *ClusterGroupUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Version: "v1",
 	})
 
+	maxConcurrency, set := os.LookupEnv("TALM_CGU_CTRL_WORKER_COUNT")
```

Comment:
Suggestion: perhaps we should declare the environment variable name in utils/constants.go and reference the constant here rather than defining the value directly. See comment below.